### DISCUSSION TITLE
[React-Aria] Nextjs Setup

### DISCUSSION THREAD

**Discussion by TheAggressive** [adobe/react-spectrum#2636] [Q&A] [answered]
I see in the docs it says to wrap your wrap your application in an SSR Provider.

my question is should you do that per component or just wrap the whole app in the _app.js?

```js
_app.js

import {SSRProvider} from '@react-aria/ssr';

function MyApp({ Component, pageProps }) {
  return ( 
      <SSRProvider>
          <Component {...pageProps} />
      </SSRProvider>
    );
}
```

**Comment by devongovett** [comment, marked as answer]
It should be around the whole app. This will guarantee that any ids that are generated are globally unique.

**Reply by TheAggressive** [reply]
Cool so in _app.js sounds about right.
